tensorflow/python/autograph/operators/py_builtins_test.py

```diff
@@ -123,6 +123,28 @@ def test_len(self):
       tl = py_builtins.len_(data_structures.tf_tensor_list_new([3, 4, 5]))
       self.assertEqual(self.evaluate(tl), 3)
 
+  def test_len_dataset(self):
+    dataset = dataset_ops.DatasetV2.from_tensor_slices([3, 2, 1])
+    self.assertEqual(py_builtins.len_(dataset), 3)
+    with self.cached_session() as sess:
+      t = py_builtins.len_(dataset)
+      self.assertAllEqual(self.evaluate(t), 3)
+
+  def test_len_dataset_infinite(self):
+    dataset = dataset_ops.DatasetV2.range(5).repeat().batch(2)
+    with self.assertRaises(errors_impl.InvalidArgumentError):
+      _ = py_builtins.len_(dataset)
+    with self.cached_session() as sess:
```

Comment:
I think this latter half of the test is redundant. Perhaps did you want to test it in graph mode? To do that, run it inside a tf.function.

Reply:
Thanks @mdanatg. Done.